### PR TITLE
implement getting comments for article

### PR DESCRIPTION
**Pull request by Aleksanrda**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -112,4 +114,78 @@ describe('API', () => {
...
+                votes: 14,
+                created_at: "2020-10-31T03:03:00.000Z"
+              };
+
+            return request(app)
+                .get("/api/articles/1/comments")
+                .expect(200)
+                .then(({ body }) => {
+                    const topBody = body.comments[0];
+
```
Comment (xViolaine) [inline __tests__/app.test.js:170]: We also want another Status 200 Test for when the ID is valid, but the comments respond with an empty array

File: app.js [hunk 2]
```diff
@@ -3,15 +3,28 @@ const app = express();
 
 const { getTopics } = require("./controllers/topic.controllers.js");
 const { getArticles } = require("./controllers/article.controllers.js");
+const { getArticleComments } = require("./controllers/comment.controllers.js");
+
+app.use(express.json());
```
Comment (xViolaine) [inline app.js:8]: We don't use app.use(express.json()); here

File: models/comment.models.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            msg: `You passed ${article_id}. Article id should be a number.`});
+    }
+
+    const selectArticleCommentsQuery = `
+    SELECT * FROM comments 
+    WHERE comments.article_id = $1;`;
+
+    return db.query(selectArticleCommentsQuery, [article_id])
+        .then((results) => {
+            console.log(results);
```
Comment (xViolaine) [inline models/comment.models.js:16]: We want to make sure to delete any unneeded console.logs :)

File: models/comment.models.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const fetchArticleComments = (article_id) => {
+    if(!parseInt(article_id)) {
+        return Promise.reject({ 
+            status: 400,
+            msg: `You passed ${article_id}. Article id should be a number.`});
+    }
+
+    const selectArticleCommentsQuery = `
+    SELECT * FROM comments 
+    WHERE comments.article_id = $1;`;
```
Comment (xViolaine) [inline models/comment.models.js:12]: We want our articles to be ordered in a specific way

File: controllers/comment.controllers.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const { fetchArticleComments } = require("../models/comment.models.js");
+
+const getArticleComments = (request, response, next) => {
+    const { article_id } = request.params;
+
+    fetchArticleComments(article_id)
+        .then(({ rows: results, rowCount}) => {
```
Comment (xViolaine) [inline controllers/comment.controllers.js:7]: If the article exists, but it has no comments, we want to send back a 200 and just send back an empty array

**Review by xViolaine — CHANGES_REQUESTED** [review]

**Review by xViolaine — APPROVED** [review]
